src/index.js

```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+module.exports = {
+  Stop: require('./circuit/hop'),
+  Dialer: require('./dialer'),
+  multicodec: require('./multicodec'),
+  tag: 'Circuit'
+}
```

Comment:
@lgierth this is an implementation detail on how `js-libp2p-swarm` is built. libp2p-circuit is a transport in the sense that there is a full conn upgrade to secio, stream-muxer and so on, but it is not a transport in the sense that you 'listen on a multiaddr' that you can pass on a Addresses.Swarm or that there is even a listener. 

The way it is coded now basically puts relayed connections on the "pipeline for incoming connections" by faking a create-listener call yields conns as if they all came from a listener. This works but it is also hard wiring things.

I rather prefer have a clear distinction in the code that there is a pipeline of things that happen for new conns (secio -> stream muxer -> identify) than having to backport and simulate whatever feature we add to https://github.com/libp2p/interface-transport. 

Note, even now there are no `.getAddrs` for this 'listener' object Relay is creating, making it fail to pass the interface definition already. This is my concern.

So answering your question, I still agree that is considered a transport for libp2p, I just don't expect it to trickle on hardwiring things on the implementation side.